File: EPI/6.6/swift/6.6.swift
// swiftlint:disable variable_name

public func removeDuplicates(_ array: inout [Int]) {
    var deduplicated = 1

    for i in stride(from: 1, to: array.count, by: 1) {
        if array[i] != array[deduplicated - 1] {
            array[deduplicated] = array[i]
            deduplicated += 1
        }
    }

    while array.count > deduplicated {
        array.removeLast()
    }
}

File: EPI/6.6/typescript/6.6.ts
export default function removeDuplicates(sortedValues: number[]): number {
    if (!sortedValues.length) {
        return 0;
    }

    let i = 0;

    for (let j = 1; j < sortedValues.length; j += 1) {
        if (sortedValues[i] !== sortedValues[j]) {
            sortedValues[i + 1] = sortedValues[j];
            i += 1;
        }
    }

    return i + 1;
}
